tests/extension/decimal/array.py

```diff
@@ -200,19 +200,15 @@ def reconstruct(
             return DecimalArray._from_sequence(x)
 
         if ufunc.nout > 1:
-            return tuple(
-                reconstruct(x)  # pyright: ignore[reportUnknownArgumentType]
-                for x in result
-            )
-        return reconstruct(result)  # pyright: ignore[reportUnknownArgumentType]
+            return tuple(reconstruct(x) for x in result)
+        return reconstruct(result)
 
     def __getitem__(self, item: ScalarIndexer | SequenceIndexer) -> Any:
         if isinstance(item, numbers.Integral):
             return self._data[item]
         # array, slice.
-        item = check_array_indexer(
-            self,
-            item,  # type: ignore[arg-type] # pyright: ignore[reportArgumentType,reportCallIssue]
+        item = check_array_indexer(  # pyright: ignore[reportCallIssue,reportUnknownVariableType]
+            self, item  # type: ignore[arg-type] # pyright: ignore[reportArgumentType]
```

Comment:
Is it possible to fix the declaration of `check_array_indexer()` so that we don't need all these ignores?